chore(consensus): Port Blob Defaults Code

alesforz opened the pull request.

jmalicevic committed:
Added maxBytes param update to e2e

jmalicevic committed:
Update ci.toml with proper blob_max_bytes value and fix check

jmalicevic committed:
fixed edge case when blobs are disabled, lowered the height they are activated at

jmalicevic committed:
Minor fixes to address initial comments

jmalicevic committed:
BlobMaxBytes update logic is the same as VE

jmalicevic committed:
Removed printf

jmalicevic committed:
Moved BlobMaxBytes var close to BlobMaxBytesUpdateHeight

jmalicevic committed:
Do not use blob in UTs

jmalicevic committed:
Fix UT to use blobs and set maxBlobBytes to 800kb

jmalicevic committed:
Fix typo

jmalicevic committed:
Applied review comments

jmalicevic committed:
Fix UTs

jmalicevic committed:
Update comment

Co-authored-by: Sergio Mena <sergio@informal.systems>

jmalicevic committed:
Update comment

Co-authored-by: Sergio Mena <sergio@informal.systems>

Commit:
fix merge oddities and run linter

Commit:
manually add changes that cherry-pick failed to add

jmalicevic commented on internal/consensus/byzantine_test.go:
```diff
@@ -378,6 +372,7 @@ func TestByzantineConflictingProposalsWithPartition(t *testing.T) {
 		require.NoError(t, err)
 
 		conR := NewReactor(css[i], true) // so we don't start the consensus states
+		conR.conS.state.ConsensusParams.Blob.MaxBytes = types.MaxBlobSizeBytes
```
This might not be needed as here by default the size is not 0, but it does not hurt.

jmalicevic commented on internal/consensus/common_test.go:
```diff
@@ -960,9 +962,9 @@ func randConsensusNetWithPeers(
 			privVal = privVals[i]
 		} else {
-			tempKeyFile, err := os.CreateTemp("", "priv_validator_key_")
+			tempKeyFile, err := os.CreateTemp(t.TempDir(), "priv_validator_key_")
```
Why these changes?

alesforz replied:
it's the linter I believe. It inserted the updated go API way of creating a temp folder

jmalicevic commented on test/e2e/pkg/manifest.go:
```diff
@@ -104,7 +104,7 @@ type Manifest struct {
 	// BlobMaxBytes specifies the maximum size in bytes of a blob. This
 	// value will be written to the genesis file of all nodes.
-	BlobMaxBytes int64 `toml:"blob_max_bytes"`
+	// BlobMaxBytes int64 `toml:"blob_max_bytes"`
```
This should be removed, I hope it is not left in the original PR . 

```suggestion
```

Commit:
delete commented out BlobMaxBytes

Commit:
deleted consensus folder mistakenly created by merge

Commit:
linting

Commit:
update go version to 1.23.8 to address vuln

Commit:
update go version in github action and docker

Commit:
Revert "update go version in github action and docker"

This reverts commit 750f4b7911dbe4e74edaeb2d326a4f4cfd7716c1.

Commit:
Revert "update go version to 1.23.8 to address vuln"

This reverts commit 9486a9e580e55c835e0030d9d193bed9c32e9ad5.

alesforz commented:
I reverted the commits bumping Go's version, because I think this should be a decision of Berachain's team (and pushed in a separate PR).

Commit:
linting

Commit:
fix incorrect automatic linting by CLI tool

Commit:
format randConsensusNetWithPeers for easier reading when debugging

Commit:
fix unit test TestReactorRecordsVotesAndBlockParts

Commit:
enable PBTS from height 1

Commit:
Map host port 5433 to postgres docker container 5432.

Fix for the `state/indexer/sink/psql` tests failing with a docker port error.

Commit:
Revert "enable PBTS from height 1"

This reverts commit 2fbe78bda3688ce45859180eb1a5a61ad5aa3c71.

alesforz commented:
e2e error below. We do enable PBTS in the `ci.toml`, so I'm not sure why this happens.

```
validator04  | E[2025-06-06|14:16:35.513] CONSENSUS FAILURE!!!    module=consensus err="PBTS has to be enabled"
stack:
goroutine 153 [running]:
    runtime/debug.Stack()
        runtime/debug/stack.go:26 +0x67
    github.com/cometbft/cometbft/internal/consensus.(*State).receiveRoutine.func2()
        github.com/cometbft/cometbft/internal/consensus/state.go:861 +0x58
    panic({0x1eed680?, 0x24b5eb0?})
        runtime/panic.go:785 +0x132
    github.com/cometbft/cometbft/state.State.MakeBlock(
        { {0xb, 0x1}, {0xc0005c247b, 0x5} },
        {0xc0005c248e, 0x2},
        0x1,
        0x0,
        { {0x0, 0x0, ...}, ... },
        ...
    )
        github.com/cometbft/cometbft/state/state.go:257 +0x685
    github.com/cometbft/cometbft/state.(*BlockExecutor).CreateProposalBlock(
        _,
        {_, _},
        _,
        { {0xb, 0x1}, {0xc0005c247b, 0x5} },
        {0xc0005c248e, 0x2},
        ...,
    )
        github.com/cometbft/cometbft/state/execution.go:142 +0x5e5
    github.com/cometbft/cometbft/internal/consensus.(*State).createProposalBlock(
        0xc000285c08,
        {0x24cab78, 0x401b260}
    )
        github.com/cometbft/cometbft/internal/consensus/state.go:1535 +0x798
    github.com/cometbft/cometbft/internal/consensus.(*State).defaultDecideProposal(
        0xc000285c08,
        0x1,
        0x0
    )
        github.com/cometbft/cometbft/internal/consensus/state.go:1336 +0xe5
    github.com/cometbft/cometbft/internal/consensus.(*State).enterPropose(
        0xc000285c08,
        0x1,
        0x0
    )
        github.com/cometbft/cometbft/internal/consensus/state.go:1312 +0x119e
    github.com/cometbft/cometbft/internal/consensus.(*State).enterNewRound(
        0xc000285c08,
        0x1,
        0x0
    )
        github.com/cometbft/cometbft/internal/consensus/state.go:1222 +0x14f0
    github.com/cometbft/cometbft/internal/consensus.(*State).handleTimeout(
        _,
        {_, _, _, _},
        {0x1, 0x0, 0x1, {0x1e286186, 0xedfd4ef42, ...}, ...}
    )
        github.com/cometbft/cometbft/internal/consensus/state.go:1088 +0xc9e
    github.com/cometbft/cometbft/internal/consensus.(*State).receiveRoutine(
        0xc000285c08,
        0x0
    )
        github.com/cometbft/cometbft/internal/consensus/state.go:925 +0xa5f
created by github.com/cometbft/cometbft/internal/consensus.(*State).OnStart in goroutine 116
    github.com/cometbft/cometbft/internal/consensus/state.go:406 +0x22b
```

Commit:
Fix testnet PBTS panic

Commit:
fix testnet BlobMaxBytes=0 panic

alesforz commented:
Fixed it. I found the issue: a sneaky merge added fields to the `Testnet` struct that were already defined in its embedded `Manifest` struct. As a result, at runtime, accessing something like `testnet.InitialHeight` read the uninitialized `Testnet.InitialHeight` instead of `Manifest.InitialHeight`. In other words, the values set on the manifest were ignored because the duplicate fields shadowed them.

alesforz commented:
Ran the nightlies. The unit tests of`networks/nightly/gen-group02-0001.toml` failed, everything else passes.

The errors:
```
FAIL: TestApp_Tx (204.06s)
    --- FAIL: TestApp_Tx/full02 (60.00s)
        e2e_test.go:64: 
            	Error Trace:	/Users/alesforz/workspace/forks/cometbft-alesforz/test/e2e/tests/app_test.go:98
            	            				/Users/alesforz/workspace/forks/cometbft-alesforz/test/e2e/tests/e2e_test.go:64
            	Error:      	Condition never satisfied
            	Test:       	TestApp_Tx/full02
            	Messages:   	submitted tx (E6EC3F4D86C9644CE3B039FBD8C369333822AEB76A5DBC517BD5F78ED84B961F) wasn't committed after 1m0s
    --- FAIL: TestApp_Tx/full03 (60.00s)
        e2e_test.go:64: 
            	Error Trace:	/Users/alesforz/workspace/forks/cometbft-alesforz/test/e2e/tests/app_test.go:98
            	            				/Users/alesforz/workspace/forks/cometbft-alesforz/test/e2e/tests/e2e_test.go:64
            	Error:      	Condition never satisfied
            	Test:       	TestApp_Tx/full03
            	Messages:   	submitted tx (68C4D6EE6D43ADC332D93A6BE1A972C14CF804F5F84E59D279D6AE6868A93547) wasn't committed after 1m0s
    --- FAIL: TestApp_Tx/validator05 (60.00s)
        e2e_test.go:64: 
            	Error Trace:	/Users/alesforz/workspace/forks/cometbft-alesforz/test/e2e/tests/app_test.go:98
            	            				/Users/alesforz/workspace/forks/cometbft-alesforz/test/e2e/tests/e2e_test.go:64
            	Error:      	Condition never satisfied
            	Test:       	TestApp_Tx/validator05
            	Messages:   	submitted tx (281CFAC88A9FF1482CD0FAF35617F04D989CB3653201A8B7B8CE99FFA292D25A) wasn't committed after 1m0s
--- FAIL: TestEvidence_Misbehavior (0.03s)
    evidence_test.go:20: 
        	Error Trace:	/Users/alesforz/workspace/forks/cometbft-alesforz/test/e2e/tests/evidence_test.go:20
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestEvidence_Misbehavior
        	Messages:   	difference between the amount of evidence produced and committed
--- FAIL: TestValidator_Propose (0.01s)
    validator_test.go:67: 
        	Error Trace:	/Users/alesforz/workspace/forks/cometbft-alesforz/test/e2e/tests/e2e_test.go:168
        	            				/Users/alesforz/workspace/forks/cometbft-alesforz/test/e2e/tests/validator_test.go:67
        	Error:      	Received unexpected error:
        	            	post failed: Post "http://127.0.0.1:5731/v1": dial tcp 127.0.0.1:5731: connect: connection refused
        	Test:       	TestValidator_Propose
--- FAIL: TestValidator_Sign (0.01s)
    validator_test.go:107: 
        	Error Trace:	/Users/alesforz/workspace/forks/cometbft-alesforz/test/e2e/tests/e2e_test.go:168
        	            				/Users/alesforz/workspace/forks/cometbft-alesforz/test/e2e/tests/validator_test.go:107
        	Error:      	Received unexpected error:
        	            	post failed: Post "http://127.0.0.1:5731/v1": dial tcp 127.0.0.1:5731: connect: connection refused
        	Test:       	TestValidator_Sign
FAIL
FAIL	github.com/cometbft/cometbft/test/e2e/tests	334.731s
FAIL
```

Commit:
Set `manifest.BlobMaxBytes` to `types.MaxBlobSizeBytes` if
`manifest.BlobMaxBytesUpdateHeight` < 0.

Without this change, if `manifest.BlobMaxBytesUpdateHeight` is -1,
`manifest.BlobMaxBytes` would be set to 0, thus making the nightlies fail
with a "BlobMaxBytes cannot be 0" error.
